syntax = "proto3";

package shop;

service ShopService {
    rpc PurchaseItem(PurchaseRequest) returns (PurchaseResponse);
    rpc MakePayment(PaymentRequest) returns (PaymentResponse);
    rpc PlaceOrder(OrderRequest) returns (OrderResponse);
    rpc GetProductInfo(ProductInfoRequest) returns (ProductInfoResponse); // Nuevo servicio
}

message PurchaseRequest {
    string product_id = 1;
    int32 quantity = 2;
}

message PurchaseResponse {
    string message = 1;
    int32 total_price = 2;
}

message PaymentRequest {
    string payment_method = 1;
    float amount = 2;
}

message PaymentResponse {
    string message = 1;
    bool success = 2;
}

message OrderRequest {
    repeated PurchaseRequest items = 1;
}

message OrderResponse {
    string order_id = 1;
    repeated PurchaseResponse purchases = 2;
}

message ProductInfoRequest {
    string product_id = 1;
}

message ProductInfoResponse {
    string name = 1;
    string description = 2;
    float price = 3;
}
